build/source/engine/checkStruc.f90

```diff
@@ -0,0 +1,227 @@
+! GNU General Public License for more details.
+!
+! You should have received a copy of the GNU General Public License
+! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+module checkStruc_module
+USE nrtype
+implicit none
+private
+public::checkStruc
+! define missing values
+integer(i4b),parameter :: missingInteger=-9999
+real(dp),parameter     :: missingDouble=-9999._dp
+contains
+
+
+ ! ************************************************************************************************
+ ! public subroutine checkStruc: check data structures 
+ ! ************************************************************************************************
+ subroutine checkStruc(err,message)
+ ! ascii utilities
+ USE ascii_util_module,only:split_line
+ ! summary of data structures
+ USE globalData,only:structInfo
+ ! metadata structures
+ USE globalData,only:time_meta,forc_meta,attr_meta,type_meta        ! metadata structures
+ USE globalData,only:prog_meta,diag_meta,flux_meta,deriv_meta       ! metadata structures
+ USE globalData,only:mpar_meta,indx_meta                            ! metadata structures
+ USE globalData,only:bpar_meta,bvar_meta                            ! metadata structures
+ ! named variables defining strructure elements
+ USE var_lookup,only:iLookTIME,iLookFORCE,iLookATTR,iLookTYPE       ! named variables showing the elements of each data structure
+ USE var_lookup,only:iLookPROG,iLookDIAG,iLookFLUX,iLookDERIV       ! named variables showing the elements of each data structure
+ USE var_lookup,only:iLookPARAM,iLookINDEX                          ! named variables showing the elements of each data structure
+ USE var_lookup,only:iLookBPAR,iLookBVAR                            ! named variables showing the elements of each data structure
+ implicit none
+ ! dummy variables
+ integer(i4b),intent(out)             :: err         ! error code
+ character(*),intent(out)             :: message     ! error message
+ ! local variables
+ integer(i4b)                         :: iStruct     ! index of data structure
+ integer(i4b),parameter               :: nStruct=size(structInfo)  ! number of data structures
+ character(len=8192)                  :: longString  ! string containing the indices defined in the structure constructor
+ character(len=32),allocatable        :: words(:)    ! vector of words extracted from the long string
+ integer(i4b)                         :: ix          ! index of the variable in the data structure
+ integer(i4b)                         :: ixTest      ! test the structure constructor = (1,2,3,...,nVar)
+ character(len=256)                   :: cmessage    ! error message of downwind routine
+ ! -----------------------------------------------------------------------------------------------------------------------------------
+ ! initialize errors
+ err=0; message="checkStruc/"
+
+ ! -----
+ ! * check that the structure constructors are correct...
```

Comment:
Some more info on how you are checking this would be helpful. The test themselves are rather cryptic and I wonder if it is even possible to fail the tests ...